operators/src/processing/time_projection/mod.rs

```diff
@@ -0,0 +1,560 @@
+use std::sync::Arc;
+
+use crate::engine::{
+    ExecutionContext, InitializedVectorOperator, Operator, QueryContext, SingleVectorSource,
+    TypedVectorQueryProcessor, VectorOperator, VectorQueryProcessor, VectorResultDescriptor,
+};
+use crate::util::Result;
+use async_trait::async_trait;
+use futures::stream::BoxStream;
+use futures::{StreamExt, TryStreamExt};
+use geoengine_datatypes::collections::{
+    FeatureCollection, FeatureCollectionInfos, FeatureCollectionModifications,
+};
+use geoengine_datatypes::primitives::{Geometry, TimeInterval};
+use geoengine_datatypes::primitives::{TimeInstance, TimeStep, VectorQueryRectangle};
+use geoengine_datatypes::util::arrow::ArrowTyped;
+use log::debug;
+use rayon::iter::{IndexedParallelIterator, IntoParallelRefIterator, ParallelIterator};
+use rayon::ThreadPool;
+use serde::{Deserialize, Serialize};
+use snafu::{ensure, ResultExt, Snafu};
+
+/// Projection of time information in queries and data
+///
+/// This operator changes the temporal validity of the queried data.
+/// In order to query all valid data, it is necessary to change the query rectangle as well.
+///
+// TODO: extend for rasters
```

Comment:
i don't think we can implement this for raster data in a similar way. Raster data will always require aggregation.

Reply:
[4c652a4](https://github.com/geo-engine/geoengine/pull/478/commits/4c652a48661bb206769e7947d0640c4f1f33c567)